ipn/ipnlocal/state_test.go

```diff
@@ -1561,6 +1561,158 @@ func TestEngineReconfigOnStateChange(t *testing.T) {
 	}
 }
 
+// TestStateMachineURLRace tests that wgengine updates arriving in the middle of
+// processing an auth URL doesn't result in the auth URL being cleared.
+func TestStateMachineURLRace(t *testing.T) {
+	runTestStateMachineURLRace(t, false)
+}
+
+func TestStateMachineURLRaceSeamless(t *testing.T) {
+	runTestStateMachineURLRace(t, true)
```

Comment:
This fails 93% of the time without the code fixes in place. I'd love to get that to 100% as well but I'm sure people would notice a test flaking 93% of the time in the future.